### PR TITLE
feat(file-list): allow a file list/dir instead of a root_dir

### PR DESCRIPTION
**Pull request by cedroyer**
## General summary

Add the ability to provide a list of files/directories instead one
Also add ability to disable unused warnings with a parameter `--no-unused`

### CODE REVIEW

File: dep_check/use_cases/interfaces.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ class Configuration:
     dependency_rules: DependencyRules = field(default_factory=dict)
     local_init: bool = False
     error_on_unused: bool = False
+    check_unused: bool = True
```
Comment (anis-campos) [inline dep_check/use_cases/interfaces.py:19]: isn't `error_on_unused` useless now ?

**Review by anis-campos — APPROVED** [review]

File: dep_check/use_cases/check.py [hunk 2]
```diff
@@ -135,7 +135,11 @@ def run(self) -> None:
         unused = all_rules.difference(self.used_rules)
         self.report_printer.print_report(errors, unused, nb_files)
 
-        if self.configuration.error_on_unused and unused:
+        if (
+            self.configuration.check_unused
+            and self.configuration.error_on_unused
+            and unused
+        ):
```
Comment (noirbee) [inline dep_check/use_cases/check.py:142]: Shouldn't `unused` be even computed if `not self.configuration.check_unused` ?

File: dep_check/use_cases/interfaces.py (hunk 1 above)
Comment (noirbee) [reply dep_check/use_cases/interfaces.py:19]: You may want the warnings but not the error. Both could probably be factored into `--unused=error|warning|ignore` but that's another story

**Review by noirbee — APPROVED** [review]
Nitpicks

File: dep_check/use_cases/interfaces.py (hunk 1 above)
Comment (cedroyer) [reply dep_check/use_cases/interfaces.py:19]: `--unused=error|warning|ignore` :+1: good idea

**Pull request merged**